### PR TITLE
Add codesandbox ci config

### PR DESCRIPTION
**Pull request by msutkowski**
Just a test to see if we can get CSB CI to build PRs 🎉

### CODE REVIEW

**Comment by msutkowski** [comment]
@kettanaito Hey Artem, would you be willing to add the CodesandboxCI app and allow it to build PRs for `msw`? You just need to [add the app as mentioned here](https://codesandbox.io/docs/ci). I'd like to get a slug for your `ws` PR over at #396 so I can setup an example for it for `rtk-query` using `ws` as a baseQuery. We use this _extensively_ with RTK... examples:

https://ci.codesandbox.io/status/rtk-incubator/rtk-query
https://ci.codesandbox.io/status/reduxjs/redux-toolkit

If you're okay with this, I can setup an example with `msw` to just build automatically in a CSB as well, like I do in the examples seen throughout these docs: https://deploy-preview-43--rtk-query-docs.netlify.app/concepts/prefetching

Thanks!!!!!!!!

File: .codesandbox/ci.json
Comment (kettanaito) [inline .codesandbox/ci.json]: Can we specify a Codesandbox document to use as a template for incremental builds?

As we have a [React sandbox](https://codesandbox.io/s/goofy-darwin-xx1c8?file=/src/index.js), we may create a new sandbox dedicated for PR previews. Unlike the React one, it would use TypeScript, plain JavaScript and have a few tests. That way we could see browser and NodeJS integrations in a single sandbox.

**Comment by kettanaito** [comment]
Hey, @msutkowski. That's an awesome suggestion!

I've added CSB App to this repository. Let me know what else is needed to have a CSB integration. 

It'd be a great idea to populate the docs with some Codesandbox examples indeed!

File: .codesandbox/ci.json
Comment (msutkowski) [reply .codesandbox/ci.json]: That's actually what I do in `rtk-query`... there is some config shenanigans we had to deal with to get the examples to run locally as well in CSB, but I could probably get both scenarios working for `msw` without too much an issue. If you look, the Svelte example uses the node template, while React uses CRA. https://github.com/rtk-incubator/rtk-query/tree/main/examples

**Comment by msutkowski** [comment]
Alright, it builds the sandbox that you linked me to, and if you check the package.json in it, it points to the build of the latest commit :). You can safely merge this, and if you want to have the examples locally in the repo, we can try and set that up as well... but it [looks like this works as expected](https://codesandbox.io/s/msw-react-reproduction-5g45m?file=/package.json) :)

File: .codesandbox/ci.json
Comment (msutkowski) [reply .codesandbox/ci.json]: Just to elaborate on 'shenanigans': it was problematic getting the tests to run and to avoid errors regarding multiple versions of React. The examples would try to reference the React/jest/etc versions in the root and throw 'multiple versions of react' errors regarding hooks usage. It took a bit of tinkering to get everything configured so that things run everywhere without problems.

**Comment by kettanaito** [comment]
This looks awesome, @msutkowski!
Before we merge I'd like for those sandboxes to contain a meaningful reproduction scenario (such as a usage and a test for that usage). Currently vanilla and TypeScript sandboxes are good, but one'd have to write usage/tests manually in order to try the library, which decreases the usefulness of this feature.

**Comment by kettanaito** [comment]
I'll try to set up such sandboxes and link them here. You can also do this, if you have time, of course. 
Thank you for the amazing work!

**Comment by kettanaito** [comment]
Hey, @msutkowski 👋 I've added a single template for now (MSW React) so we could merge this change straight away. I'm looking forward to other sandboxes you may create when you have time! 

Will rebase the branch and merge. Once more thank you for this awesome addition!

**Review by kettanaito — APPROVED** [review]

**Pull request merged**
